### PR TITLE
reportlab: Move Pillow into propagatedBuildInputs

### PR DESCRIPTION
**Pull request by hacker1024**
###### Description of changes

Pillow needs to be available at runtime in the reportlab derivation, as it is a Python package.
This PR moves it into `propagatedBuildInputs`.

This fixes things that depend on reportlab like the Kyocera print filter, which otherwise causes a "filter error" in CUPS when the Python script crashes.

@FRidh

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by marsam** [comment]
@GrahamcOfBorg eval

File: pkgs/development/python-modules/reportlab/default.nix [hunk 1]
```diff
@@ -25,7 +25,9 @@ in buildPythonPackage rec {
 
   nativeCheckInputs = [ glibcLocales ];
 
-  buildInputs = [ ft pillow ];
+  buildInputs = [ ft ];
+
```
Comment (panicgh) [inline pkgs/development/python-modules/reportlab/default.nix:29]: The EditorConfig check fails due to the trailing spaces:

```suggestion

```

**Review by panicgh — CHANGES_REQUESTED** [review]
I tested it with cups-kyodialog and it fixes the filter issue. The error fixed in this PR can be reproduced on `master` with:

```console
$ nix-shell -I nixpkgs=. -p 'python3.withPackages(ps: with ps; [ reportlab ])' --run 'python3 -c "from reportlab.pdfgen import canvas"'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/nix/store/7rkm2qzgdyvrrzx0vasr8gj1w09zvc8s-python3-3.10.10-env/lib/python3.10/site-packages/reportlab/pdfgen/canvas.py", line 16, in <module>
    from reportlab import rl_config
  File "/nix/store/7rkm2qzgdyvrrzx0vasr8gj1w09zvc8s-python3-3.10.10-env/lib/python3.10/site-packages/reportlab/rl_config.py", line 45, in <module>
    _DEFAULTS=_defaults_init()
  File "/nix/store/7rkm2qzgdyvrrzx0vasr8gj1w09zvc8s-python3-3.10.10-env/lib/python3.10/site-packages/reportlab/rl_config.py", line 13, in _defaults_init
    from reportlab.lib.utils import rl_exec
  File "/nix/store/7rkm2qzgdyvrrzx0vasr8gj1w09zvc8s-python3-3.10.10-env/lib/python3.10/site-packages/reportlab/lib/utils.py", line 15, in <module>
    from PIL import Image
ModuleNotFoundError: No module named 'PIL'
```

Result of `nixpkgs-review pr 216761` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>odoo</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>python311Packages.cnvkit</li>
  </ul>
</details>
<details>
  <summary>43 packages built:</summary>
  <ul>
    <li>actdiag (python310Packages.actdiag)</li>
    <li>asciidoc-full-with-plugins</li>
    <li>blockdiag (python310Packages.blockdiag)</li>
    <li>cups-kyodialog</li>
    <li>cve-bin-tool</li>
    <li>hocr-tools (python310Packages.hocr-tools)</li>
    <li>hplip</li>
    <li>hplipWithPlugin</li>
    <li>maigret</li>
    <li>nwdiag (python310Packages.nwdiag)</li>
    <li>ocrfeeder</li>
    <li>ocrmypdf (python310Packages.ocrmypdf)</li>
    <li>paperless-ngx</li>
    <li>python310Packages.cnvkit</li>
    <li>python310Packages.easy-thumbnails</li>
    <li>python310Packages.reportlab</li>
    <li>python310Packages.rmrl</li>
    <li>python310Packages.rst2pdf</li>
    <li>seqdiag (python310Packages.seqdiag)</li>
    <li>python310Packages.sphinxcontrib-actdiag</li>
    <li>python310Packages.sphinxcontrib-blockdiag</li>
    <li>python310Packages.sphinxcontrib-nwdiag</li>
    <li>python310Packages.sphinxcontrib-seqdiag</li>
    <li>python310Packages.svglib</li>
    <li>python310Packages.xhtml2pdf</li>
    <li>python311Packages.actdiag</li>
    <li>python311Packages.blockdiag</li>
    <li>python311Packages.easy-thumbnails</li>
    <li>python311Packages.hocr-tools</li>
    <li>python311Packages.nwdiag</li>
    <li>python311Packages.ocrmypdf</li>
    <li>python311Packages.reportlab</li>
    <li>python311Packages.rmrl</li>
    <li>python311Packages.rst2pdf</li>
    <li>python311Packages.seqdiag</li>
    <li>python311Packages.sphinxcontrib-actdiag</li>
    <li>python311Packages.sphinxcontrib-blockdiag</li>
    <li>python311Packages.sphinxcontrib-nwdiag</li>
    <li>python311Packages.sphinxcontrib-seqdiag</li>
    <li>python311Packages.svglib</li>
    <li>python311Packages.xhtml2pdf</li>
    <li>rmfuse</li>
    <li>sasview</li>
  </ul>
</details>

**Comment by panicgh** [comment]
The checks now pass (great!), but wouldn't it be better to squash the two commits into a single one?

Also, I think the commit message "Add trailing newline to default.nix" is misleading, because the complaint from the bot was not because of newlines, but because of whitespace from indentation on an empty line. Just your initial commit without the indentation on the empty line would be enough.

**Pull request merged**